rest/route/host_agent.go

```diff
@@ -483,6 +485,39 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			return nil, true, nil
 		}
 
+		if currentHost.Distro.SingleTaskDistro {
+			// If next task exists and the distro is a single task distro, check if the task is allowed on the distro.
+			allowedTasks, err := validator.GetAllowedSingleTaskDistroTasksForProject(ctx, nextTask.Project)
+			if err != nil {
+				grip.Alert(message.WrapError(err, errMsg))
```

Comment:
Is `errMsg` supposed to be set here? The `errMsg` above seems to be irrelevant to this error.